app/src/components/SectionPicker.tsx

```diff
@@ -7,46 +7,63 @@ import { getSearchByStringExample } from "@constants/frontend";
 
 interface Props {
     soc: SOC_Generic;
-    searchText: string;
-    setSearchText: (searchText: string) => void;
 }
 
 export default function SectionPicker(props: Props) {
     const [searchByString, setSearchByString] = useState<string>(
         getSearchByString(SearchBys[0]),
     );
+    const [searchText, setSearchText] = useState("");
     const searchBy = getSearchBy(searchByString);
 
     /* https://github.com/apollographql/apollo-client/issues/9583 */
     const [abortRef, setAbortRef] = useState(new AbortController());
-    const { isFetching, data: courses } = useQuery<Course[]>({
+    const {
+        isFetching,
+        data: courses,
+        refetch: fetchCourses,
+    } = useQuery<Course[]>({
         initialData: [],
-        queryKey: [props.searchText], // Re-fetch when searchText is updated
+        // queryKey: [props.searchText], // Re-fetch when searchText is updated
         queryFn: () => {
             setAbortRef(new AbortController());
             return props.soc instanceof SOC_API
-                ? props.soc.fetchSearchCourses(
-                      searchBy,
-                      props.searchText,
-                      abortRef,
-                  )
-                : props.soc.searchCourses(searchBy, props.searchText);
+                ? props.soc.fetchSearchCourses(searchBy, searchText, abortRef)
+                : props.soc.searchCourses(searchBy, searchText);
         },
-        enabled: !!props.searchText, // Prevents query when searchText is empty
+        enabled: false,
+        // enabled: !!props.searchText, // Prevents query when searchText is empty
         notifyOnChangeProps: ["data", "isFetching"], // Must re-render on isFetching change to update cursor
         refetchOnWindowFocus: false, // Turned off to prevent queries while debugging using console
         refetchOnReconnect: false, // Not needed
     });
-    console.log(isFetching ? "Fetching courses..." : "Not fetching.");
+    // console.log(isFetching ? "Fetching courses..." : "Not fetching.");
+
+    const searchCourse = () => {
+        fetchCourses();
+    };
 
     // TODO: make it clearer when not fetching and there are no search results
-    const displayCourses = (
-        isFetching
-            ? props.soc instanceof SOC_API
-                ? props.soc.searchCourses(searchBy, props.searchText)
-                : []
-            : courses ?? []
-    ).slice(0, 30);
+    // const displayCourses = (
+    //     isFetching
+    //         ? props.soc instanceof SOC_API
+    //             ? props.soc.searchCourses(searchBy, props.searchText)
+    //             : []
+    //         : courses ?? []
+    // ).slice(0, 30);
+
+    const renderCourses = () => {
```

Comment:
The purpose of defining functions is is almost entirely re-usability.

We call this function once. So...

We can just get the courses and even sleeker like this:
```tsx
let coursesToDisplay = <p>Loading...</p>;
if (!isFetching)
    coursesToDisplay = (
        <MultipleCourseDisplay courses={courses ?? []} />
);
```
We don't really need to limit courses to display anymore (removed `.slice(...)`).

Reply:
Also, both your implementation and my suggestion doesn't show courses that match the query while refetching (which is useful if the course the user is looking for was already fetched).

For example, if you query "MAC", then wait till they are displayed. Then you query "MAC231" you can display "MAC2311" because we already fetched and cached it! However, you should still fetch in the background.

Reply:
Good idea. I think React Query might have something like, I'll look into it.